### DISCUSSION TITLE
Support `with_front` option during CPT registration for custom permalinks

### DISCUSSION THREAD

**Discussion by blakewilson** [wpengine/atlas-content-modeler#457] [Ideas]
When using a custom permalink structure such as `/posts/%postname%/`, permalinks for ACM posts include the `/posts/` prefix.

For example, say I have a "rabbits" model. Rabbits are inherently different than "posts". However, the permalink I get for my newly created rabbit is:
http://headless.local/posts/rabbit/333

When I would expect it to be:
http://headless.local/rabbit/333

By supporting the `with_front` [`register_post_type()`](https://developer.wordpress.org/reference/functions/register_post_type/) option, this permalink structure can be supported.

I believe this becomes increasingly important when dealing with frontend ui frameworks like Faust.js/Next.js/etc. that use file system based routing, as it would require you to have the following directory structure:

```
my-app/
├─ node_modules/
├─ src/
│  ├─ posts/
│  │  ├─ [postSlug].js
│  │  ├─ rabbit/
│  │  │ ├─ [rabbitSlug].js
├─ package.json
├─ README.md
```

Instead of:

```
my-app/
├─ node_modules/
├─ src/
│  ├─ posts/
│  │  ├─ [postSlug].js
│  ├─ rabbit/
│  │  ├─ [rabbitSlug].js
├─ package.json
├─ README.md
```

Matching WP permalink structure with your frontend url structure is valuable when considering sitemaps, and other features.

**Comment by nickcernis** [comment]
Great idea! We're testing a fix for this in https://github.com/wpengine/atlas-content-modeler/pull/468. Screenshots available in that PR.

**Comment by nickcernis** [comment]
ACM 0.15.0 is out now and supports this feature via a setting on the edit model page.

Untick “use front permalink base” and your ACM posts will no longer be prefixed by `/posts/` if your permalink structure is `/posts/%postname%/`.

Note that you will also need to re-export blueprints after making this change. (The `with_front` option is stored as a property of the model.)

Also note that the setting does not appear on the create new model form by design — only on the edit model form.

Existing models and new models are unaffected — they use front permalinks by default unless edited to remove them, consistent with WordPress defaults.

<img width="562" alt="Screenshot 2022-03-24 at 17 22 43" src="https://user-images.githubusercontent.com/647669/159963209-76cd4715-3011-4bd1-b224-d5029cc6d9c0.png">

**Reply by blakewilson** [reply]
Awesome!! Well done, and thanks for getting this in so quickly 🚀
